crowdfunding/projects/models.py

```diff
@@ -1,3 +1,62 @@
 from django.db import models
+from django.contrib.auth import get_user_model
+
+User = get_user_model() #give me the user model that you set in settings
 
 # Create your models here.
+class Project(models.Model):
+    title = models.CharField(max_length=100)
+    description = models.TextField()
+    goal = models.IntegerField()
+    image = models.URLField()
+    is_open = models.BooleanField()
+    date_created = models.DateTimeField(auto_now_add=True) #always use auto_now_add=True with date_created = models.DateTimeField
+    owner = models.ForeignKey(
+        User, 
+        on_delete=models.CASCADE, 
+        related_name='owner_projects'
+        )
+    
+    # liked_by = models.ManyToManyField(
+    #     User,
+    #     related_name='liked_projects'
+    # )
+    @property
+    def total(self):
+        return self.pledges.aggregate(sum=models.Sum('amount'))['sum']
+
+class Pledge(models.Model):
+    amount = models.IntegerField()
+    comment = models.CharField(max_length=200)
+    anonymous = models.BooleanField()
+    project = models.ForeignKey(Project, on_delete=models.CASCADE, related_name="pledges")
+    supporter_public = models.CharField(max_length=200)
+    supporter_private= models.ForeignKey(
+    User,
+    on_delete=models.CASCADE,  # using CASCADE so if you delete the project, then it will delete the pledges
+    related_name='supporter_private_pledges'
+    )
```

Comment:
oh hello!
you've created a supporter public and private to split how you attach the pledge and how you display the pledge. 
really interesting, a novel approach! 
I would think that it would be better to handle this in the serializer rather than at the point of saving to the database, what would happen if you switched from being an anonymous pledge to not one or the other way? 
nice work though!